libzeth/circuits/circuit_wrapper.tcc

```diff
@@ -27,14 +27,33 @@ circuit_wrapper<
     NumOutputs,
     TreeDepth>::circuit_wrapper()
 {
+    // Allocate a single public variable to hold the hash of the public
+    // joinsplit inputs. The public joinsplit inputs are then allocated
+    // immediately following this.
+    public_data_hash.allocate(pb, "public_data_hash");
+    pb.set_input_sizes(1);
+
     // Joinsplit gadget internally allocates its public data first.
     // TODO: joinsplit_gadget should be refactored to be properly composable.
     joinsplit = std::make_shared<joinsplit_type>(pb);
     const size_t num_public_elements = joinsplit->get_num_public_elements();
-    pb.set_input_sizes(num_public_elements);
+
+    // Populate public_data_hash to represent the joinsplit public data. Skip
+    // the first 2 variable (the constant 1 and the hash), and use the
+    // following num_public_elements variables.
+    public_data.reserve(num_public_elements);
+    for (size_t i = 0; i < num_public_elements; ++i) {
+        public_data.emplace_back(i + 2);
```

Comment:
> To which extend is this the "public data"?
> This rather seems to represent the vector of indices pointing to the wires that are treated as public data, hence I would rename this variable (i.e. public_data) to something like public_data_indices that seems to better track the intent here, otherwise this is misleading.

I'm not sure I follow.  All of these variables are "indices" in that sense.  So this is `public_data` in the same way as, say, `joinsplit::merkle_root` is the variable holding the Merkle root - we don't call it `joinsplit::merkle_root_index`.

Reply:
Just to note, `public_data` here is an array of `pb_variables`, not reuslt `size_t`.  Could that be the source of confusion?  The array has to be initialized, using the variable indices, because joinsplit allocates everything internally for now (see comment above about making it more composible).

Reply:
Thanks @dtebbs - got confused. That makes sense now.